### PR TITLE
KEYCLOAK-7703 Documentation for HierarchicalPathBasedKeycloakConfigResolver

### PR DESCRIPTION
**Pull request by hmlnarik**
Documentation for keycloak/keycloak#5319

### CODE REVIEW

File: securing_apps/topics/oidc/java/fuse7/classic-war.adoc [hunk 1]
```diff
@@ -69,11 +69,12 @@ For example:
...
 [[_fuse7_config_external_adapter]]
-====== Configuring the External Adapter
+====== Configuration Resolvers
 
-If you do not want the `keycloak.json` adapter configuration file to be bundled inside your WAR application, but instead made available externally and loaded based on naming conventions, use this configuration method.
-
-To enable the functionality, add this section to your `/WEB_INF/web.xml` file:
+The `keycloak.json` adapter configuration file can be stored inside a bundle,
+which is default behaviour, or in a directory on a filesystem. To specify the
+actual source of the configuration file, set `keycloak.config.resolver` deployment parameter to the desired configuration resolver class.
```
Comment (matthewhelmke) [inline securing_apps/topics/oidc/java/fuse7/classic-war.adoc]: Should read "...set the `keycloak.config.resolver` deployment..."

File: securing_apps/topics/oidc/java/fuse7/classic-war.adoc [hunk 2]
```diff
@@ -83,8 +84,31 @@ To enable the functionality, add this section to your `/WEB_INF/web.xml` file:
 </context-param>
 ----
 
-That component uses `keycloak.config` or `karaf.etc` java properties to search for a base folder to locate the configuration.
-Then inside one of those folders it searches for a file called `<your_web_context>-keycloak.json`.
+The available resolvers for `keycloak.config.resolver` are the following ones:
+
```
Comment (matthewhelmke) [inline securing_apps/topics/oidc/java/fuse7/classic-war.adoc]: I suggest: "The following resolvers are available for `keycloak.config.resolver`:

File: securing_apps/topics/oidc/java/fuse7/classic-war.adoc [hunk 3]
```diff
@@ -83,8 +84,31 @@ To enable the functionality, add this section to your `/WEB_INF/web.xml` file:
...
+org.keycloak.adapters.osgi.BundleBasedKeycloakConfigResolver::
+  Default resolver that looks up for the configuration file inside the bundle it
+  is configured from. By default, it loads file named `WEB-INF/keycloak.json` but this file name can be configured via `configLocation` property.
```
Comment (matthewhelmke) [inline securing_apps/topics/oidc/java/fuse7/classic-war.adoc]: I am unclear what is being said in "Default resolver that looks up for the configuration file inside the bundle it is configured from."

Are you trying to say this? "This is the default resolver, which requires a configuration file. By default, it loads..."

File: securing_apps/topics/oidc/java/fuse7/classic-war.adoc [hunk 4]
```diff
@@ -83,8 +84,31 @@ To enable the functionality, add this section to your `/WEB_INF/web.xml` file:
...
+
+org.keycloak.adapters.osgi.PathBasedKeycloakConfigResolver::
+  It searches for a file called `<your_web_context>-keycloak.json` inside a folder
```
Comment (matthewhelmke) [inline securing_apps/topics/oidc/java/fuse7/classic-war.adoc]: I suggest starting this with "This resolver searches for a file..."

File: securing_apps/topics/oidc/java/fuse7/classic-war.adoc [hunk 5]
```diff
@@ -83,8 +84,31 @@ To enable the functionality, add this section to your `/WEB_INF/web.xml` file:
...
+  that is specified by `keycloak.config` system property. If `keycloak.config` is
+  not set, `karaf.etc` system property is used instead.
++
+For example, if your web application is deployed into context `my-portal`, then
+  your adapter configuration would be loaded either from the 
+  `${keycloak.config}/my-portal-keycloak.json` file, or from `${karaf.etc}/my-portal-keycloak.json`.
+
+org.keycloak.adapters.osgi.HierarchicalPathBasedKeycloakConfigResolver::
+  Similar to `PathBasedKeycloakConfigResolver` above, where for given URI path,
```
Comment (matthewhelmke) [inline securing_apps/topics/oidc/java/fuse7/classic-war.adoc]: I suggest starting with a parallel phrase, such as "This resolver is similar to..."

**Review by matthewhelmke — CHANGES_REQUESTED** [review]
Overall, this is clear and easy to read and understand. I have only some small changes to request.

**Comment by hmlnarik** [comment]
Thanks @matthewhelmke for your comments!

**Review by pdrozd — APPROVED** [review]

**Review by matthewhelmke — APPROVED** [review]

**Pull request merged**